### PR TITLE
Clean up imports

### PR DESCRIPTION
**Pull request by ccoors**

### CODE REVIEW

**Comment by marijnvdwerf** [comment]
Ah, I see I missed the cleaned imports in `src/openrct2/network/Network.cpp`

**Comment by AaronVanGeffen** [comment]
That's a lot of unnecessary includes… Nice work!

File: src/openrct2/network/NetworkServerAdvertiser.cpp [hunk 1]
```diff
@@ -229,14 +226,18 @@ class NetworkServerAdvertiser : public INetworkServerAdvertiser
...
             sint32 hexCharIndex = util_rand() % Util::CountOf(hexChars);
             key[i] = hexChars[hexCharIndex];
         }
-        key[Util::CountOf(key) - 1] = 0;
-        return key;
+        key[TWITCH_KEY_LENGTH] = 0;
+
+        std::string result { key };
+        delete[] key;
+        return result;
```
Comment (alanbrady) [inline src/openrct2/network/NetworkServerAdvertiser.cpp]: This seems good, but maybe deserves a separate commit?

File: src/openrct2/network/TcpSocket.cpp [hunk 2]
```diff
@@ -40,12 +39,10 @@
     #endif
     #define FLAG_NO_PIPE 0
 #else
-    #include <errno.h>
+
```
Comment (alanbrady) [inline src/openrct2/network/TcpSocket.cpp]: Extra new line?

File: src/openrct2/Context.cpp [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 #include <exception>
 #include <memory>
-#include <string>
+
```
Comment (alanbrady) [inline src/openrct2/Context.cpp]: Extra new line?

File: src/openrct2/core/FileScanner.cpp [hunk 4]
```diff
@@ -23,9 +23,8 @@
 
 #if defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
     #include <dirent.h>
-    #include <sys/types.h>
     #include <sys/stat.h>
-    #include <unistd.h>
+
```
Comment (alanbrady) [inline src/openrct2/core/FileScanner.cpp]: Extra new line?

File: src/openrct2/network/NetworkServerAdvertiser.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/network/NetworkServerAdvertiser.cpp]: Its probably better to use more of std::unique_ptr to begin with.

Comment (marijnvdwerf) [reply src/openrct2/network/NetworkServerAdvertiser.cpp]: Ah yes, this should be in the PR. Must've missed this while splitting them up

File: src/openrct2/network/NetworkKey.cpp [hunk 5]
```diff
@@ -19,10 +19,8 @@
 #include <vector>
 #include <openssl/evp.h>
 #include <openssl/pem.h>
-#include <openssl/rsa.h>
```
Comment (janisozaur) [inline src/openrct2/network/NetworkKey.cpp]: Needed for `RSA_check_key`

File: src/openrct2/network/NetworkServerAdvertiser.cpp [hunk 6]
```diff
@@ -232,8 +228,11 @@ class NetworkServerAdvertiser : public INetworkServerAdvertiser
         char key[17];
         for (sint32 i = 0; i < 16; i++)
         {
-            sint32 hexCharIndex = util_rand() % Util::CountOf(hexChars);
-            key[i] = hexChars[hexCharIndex];
+            {
```
Comment (janisozaur) [inline src/openrct2/network/NetworkServerAdvertiser.cpp]: no need for this

File: src/openrct2/network/NetworkServerAdvertiser.cpp [hunk 7]
```diff
@@ -232,8 +228,11 @@ class NetworkServerAdvertiser : public INetworkServerAdvertiser
...
+                sint32 hexCharIndex = util_rand() % Util::CountOf(hexChars);
+                key[i] = hexChars[hexCharIndex];
+                key[i] = hexChars[hexCharIndex];
```
Comment (janisozaur) [inline src/openrct2/network/NetworkServerAdvertiser.cpp]: this also looks bogus, probably a typo

File: src/openrct2/network/TcpSocket.cpp [hunk 8]
```diff
@@ -18,7 +18,6 @@
 
 #include <chrono>
 #include <future>
-#include <thread>
```
Comment (janisozaur) [inline src/openrct2/network/TcpSocket.cpp]: This needs to stay: we use `std::thread` in here

File: src/openrct2/network/TcpSocket.cpp [hunk 9]
```diff
@@ -40,12 +39,9 @@
     #endif
     #define FLAG_NO_PIPE 0
 #else
-    #include <errno.h>
```
Comment (janisozaur) [inline src/openrct2/network/TcpSocket.cpp]: this needs to stay for `errno`

File: src/openrct2/network/TcpSocket.cpp [hunk 10]
```diff
@@ -40,12 +39,9 @@
...
     #include <arpa/inet.h>
     #include <netdb.h>
     #include <netinet/tcp.h>
-    #include <netinet/in.h>
-    #include <sys/socket.h>
```
Comment (janisozaur) [inline src/openrct2/network/TcpSocket.cpp]: needs to stay for `sockaddr_storage`

File: src/openrct2/object/LargeSceneryObject.cpp [hunk 11]
```diff
@@ -15,10 +15,8 @@
 #pragma endregion
 
 #include "../core/IStream.hpp"
-#include "../core/Memory.hpp"
```
Comment (janisozaur) [inline src/openrct2/object/LargeSceneryObject.cpp]: This include is removed in many places, but it's still in use, just a few lines down in `LargeSceneryObject::~LargeSceneryObject()`. Please only remove when file does not make use of any of `Memory::`

**Comment by janisozaur** [comment]
I think there are too many includes removed in some places. It all still works, because other headers still include removed includes, so there's little overall change.

Removing system headers is especially dangerous as it can silently break other systems – we only test so many configurations on travis.

**Comment by Gymnasiast** [comment]
If a file uses something, it should include it. A file should get its own kitchen sink, rather than relying on higher up files to include it for them.

**Comment by ccoors** [comment]
Okay, so.. close?

**Comment by Gymnasiast** [comment]
@ccoors Well, there are plenty of unused includes. What I'm saying is that you should not remove includes of stuff that is actually used - memory is one example.

File: src/openrct2/drawing/NewDrawing.cpp [hunk 12]
```diff
@@ -23,12 +23,9 @@
 #include "../paint/Painter.h"
 #include "IDrawingContext.h"
 #include "IDrawingEngine.h"
-#include "NewDrawing.h"
```
Comment (Gymnasiast) [inline src/openrct2/drawing/NewDrawing.cpp]: Are you sure about this?

Comment (ccoors) [reply src/openrct2/drawing/NewDrawing.cpp]: No. I'm going to close this PR as it's impossible to thoroughly check all includes manually. Sorry :/ But I learned something :)

**Comment by janisozaur** [comment]
@ccoors ok, cleaning up includes is tricky business. It's never _that_ easy. there were still some valuable things in this PR and I will try salvaging them later.

Your contribution is still welcome, even if this PR did not end up getting merged.
